pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -73,6 +81,39 @@ func (r *Client) RemoveSnapshots(vmRef ref.Ref, precopies []planapi.Precopy) (er
 	return
 }
 
+//
+// Check if a snapshot is ready to transfer, to avoid importer restarts.
+func (r *Client) CheckSnapshotReady(vmRef ref.Ref, snapshot string) (ready bool, err error) {
+	correlationID, err := r.getSnapshotCorrelationID(vmRef, &snapshot)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	jobs, err := r.getJobs(correlationID)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	if len(jobs) > 1 {
+		err = liberr.New("Multiple jobs found for correlation ID %s", correlationID)
+		return
+	} else if len(jobs) < 1 {
+		err = liberr.New("No jobs found for correlation ID %", correlationID)
+		return
+	}
+	job := jobs[0]
```

Comment:
I'd check all return jobs, this is what we usually do when we use correlation id to check for completion

Reply:
I changed this around a bit to consider the status of all the listed jobs.